### PR TITLE
Logstash 5.x compatible

### PR DESCRIPTION
**Pull request by iMelnik**
General idea is to depend on 'logstash-core-plugin-api' instead of 'logstash-core' to be compatible with logstash 5.x
I've done some rubocop cleanup and added Dockerfile for running this plugin in Docker + version bump.

Did not touch tests so far

### CODE REVIEW

File: lib/logstash/inputs/google_pubsub.rb [hunk 1]
```diff
@@ -16,65 +16,63 @@
...
+  config :project_id, validate: :string, required: true
 
   # Google Cloud Pub/Sub Topic and Subscription.
   # Note that the topic must be created manually with Cloud Logging
   # pre-configured export to PubSub configured to use the defined topic.
   # The subscription will be created automatically by the plugin.
-  config :topic, :validate => :string, :required => true
-  config :subscription, :validate => :string, :required => true
-  config :max_messages, :validate => :number, :required => true, :default => 5
+  config :topic, validate: :string, required: true
```
Comment (suyograo) [inline lib/logstash/inputs/google_pubsub.rb]: Can you revert these changes? Not sure why these changes are required

File: lib/logstash/inputs/google_pubsub.rb [hunk 2]
```diff
@@ -16,65 +16,63 @@
...
+      topic=#{@topic}, subscription=#{@subscription}")
     @topic = "projects/#{@project_id}/topics/#{@topic}"
     @subscription = "projects/#{@project_id}/subscriptions/#{@subscription}"
     @subscription_exists = false
 
     # TODO(erjohnso): read UA data from the gemspec
     @client = Google::APIClient.new(
-      :application_name => 'logstash-input-google_pubsub',
-      :application_version => '0.9.0'
+      application_name: 'logstash-input-google_pubsub',
```
Comment (suyograo) [inline lib/logstash/inputs/google_pubsub.rb]: Again, we should use `:key` syntax, not the other way to be consistent with LS

**Comment by suyograo** [comment]
@iMelnik sorry for the delay. Can you address the concerns I raised above?

**Comment by suyograo** [comment]
I actually pushed a commit with these fixes and some others as well. Closing this PR https://github.com/logstash-plugins/logstash-input-google_pubsub/commit/8dcfa0e36e948d89674596e0e0d96f3c420335ad

File: lib/logstash/inputs/google_pubsub.rb (hunk 1 above)
Comment (iMelnik) [inline lib/logstash/inputs/google_pubsub.rb]: This is a change to ruby 1.9 hash syntax.
Should not harm anything
